fetch.bs

```diff
@@ -8584,10 +8593,67 @@ method steps are:
         https://github.com/whatwg/dom/issues/1031#issuecomment-1233206400 -->
   </ol>
 
+ <li><p>Let <var>hasUploadListeners</var> be false.
+
+ <li><p>Let <var>requestObserver</var> be null.
+
+ <li><p>Let <var>responseObserver</var> be null.
+
+ <li>
+  <p>If <var>init</var>["{{RequestInit/observer}}"] <a for=map>exists</a>, then:
+
+  <ol>
+   <li><p>Let <var>observerCallback</var> be <var>init</var>["{{RequestInit/observer}}"].
+
+   <li><p>Set <var>requestObserver</var> to a {{FetchObserver}}.
+
+   <li><p>Set <var>responseObserver</var> to a {{FetchObserver}}.
+
+   <li><p>Let <var>args</var> be « <var>requestObserver</var>, <var>responseObserver</var> ».
+
+   <li><p>[=invoke|Invoke=] <var>observerCallback</var> with <var>args</var>
+   and <code>"rethrow"</code>. If this throws an exception, <a for=/>reject</a> <var>p</var> with it
+   and return <var>p</var>.
+
+   <li><p>If one or more <a event><code>progress</code></a> event listeners were added to
+   <var>requestObserver</var>, then set <var>hasUploadListeners</var> to true.
```

Comment:
Does this mean there's a specific time-window where listeners can be added? Isn't that… bad?